cmd/osbuild-pipeline/main.go

```diff
@@ -101,8 +102,13 @@ func main() {
 
 	repos := make([]rpmmd.RepoConfig, len(composeRequest.Repositories))
 	for i, repo := range composeRequest.Repositories {
+		repo_name := repo.Name
+		// ensure that the name is always set to a sane value
+		if repo_name == "" {
+			repo_name = fmt.Sprintf("repo-%d", i)
```

Comment:
Should we ensure the name is unique?

Reply:
Isn't it unique because the name contains the index of the repository?

Reply:
I think that Tom mainly means that the name provided in the repo definitions as an input may not be unique. I'm OK with adding check to ensure this on input (if DNF does not fail in this case, which I didn't test), but don't think that this should be necessarily taken as a valid case and workarounded in the code to use indexes instead. Alternatively a compromise would be to use the Repo name, but keep the index and combine these two values to make it fool-proof.